source/common/http/conn_manager_impl.h

```diff
@@ -423,6 +426,7 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     absl::optional<Router::RouteConstSharedPtr> cached_route_;
     absl::optional<Upstream::ClusterInfoConstSharedPtr> cached_cluster_info_;
     DownstreamWatermarkCallbacks* watermark_callbacks_{nullptr};
+    MetadataMapVector request_metadata_map_vector_;
```

Comment:
Fixed. Thanks!

Reply:
2 things for thought:
1) Related to some of my other comments, can we potentially just be directly storing pointers to metadata maps so that we can avoid copies?
2) I'm wondering if we want to actually have some type of metadata storage structure that is created on demand? So few people will be using this feature that I'm wondering if it's worth the memory and small init cost in the common case? Thoughts?

Reply:
To answer 1: Storing metadata as unique_ptr<MetadataMap> instead of MetadataMapVector (aka std::vector<unique_ptr<MetadataMap>>) will merge two metadata with the same key value into one. This may cause some problem when multiple copies of the same metadata are expected, for example, for logging purpose.

To answer 2: I think that's good suggestion. Do you mind I make that change in a followup PR? Added a TODO.

Reply:
> To answer 2: I think that's good suggestion. Do you mind I make that change in a followup PR? Added a TODO.

I think that's OK, but will you do it relatively quickly? I think we need to be mindful of the performance/memory impact of rarely used code in the hottest paths.

Reply:
I see. Fixed in this PR.